SRE-76 move PHP classes from config to app

mixoteric opened the pull request:
https://wikia-inc.atlassian.net/browse/SRE-76

move PHP classes from config to app

* class FullRequestLogger from config/Logging.php
* class AuditLog from config/AuditLog.php
* $wgHooks assignments from CommonExtensions

https://github.com/Wikia/config/pull/2355

@drozdo @ljagiello @TK-999 @macbre

Commit:
SRE-76 move PHP classes from config to app

* class FullRequestLogger from config/Logging.php
* class AuditLog from config/AuditLog.php
* $wgHooks assignments from CommonExtensions

Commit:
SRE-76 keep original order in $wgHooks

Commit:
SRE-76 remove debug

mszabo-wikia commented on includes/wikia/FullRequestLogger.class.php:
```diff
@@ -0,0 +1,166 @@
+<?php
+
+class FullRequestLogger {
```
Let's verify if this is still needed. In the era of global auth pages, `UserLoginSpecialController` is no longer used, and the Digital Ocean IPs may well be out of date. We might also have other sources for such access logs.

ljagiello replied:
Why do we even have hardcoded IPs there? It sounds wrong.

ljagiello replied:
It looks like it was introduced back in security incident days I think we should do at CDN level not inside application.

mixoteric replied:
Okay, I will review it and remove if no longer needed.

mszabo-wikia commented on includes/wikia/AuditLog.class.php:
```diff
@@ -0,0 +1,216 @@
+	}
+
+	static public function onAddNewAccount() {
+		return true;
```
redundant hook handler

Commit:
SRE-76 retire FullRequestLogger

ghost commented:
I have removed `FullRequestLogger` completely, we are going to have ca. 3 million entries per day less in our logs. :)

Commit:
SRE-76 AuditLog remove noop code

Commit:
SRE-76 AuditLog remove obsolete hooks

Commit:
SRE-76 AuditLog: add class name to log messages

I am considering removing this code. The class name will make
searching through logs easier.

ghost commented:
Removing obsolete hooks from `CheckUser` is a bonus.
